File: backend/app/schemas/dienststunden.py
from datetime import date

from pydantic import BaseModel, Field


class DienststundenErfassen(BaseModel):
    funktion_id: int
    stunden: float = Field(gt=0)
    datum: date
    ohne_pin: bool = False


class DienststundenEintragOut(BaseModel):
    id: int
    person_id: int
    person_name: str
    funktion_id: int
    funktion_name: str
    stunden: float
    datum: date
    ohne_pin: bool


class DienststundenStempelInfo(BaseModel):
    """Öffentlicher Kontext für das Dienststunden-Stempel-Poster (QR-Ziel)."""

    funktion_id: int
    funktion_name: str
    aktiv: bool


class DienststundenSummeOut(BaseModel):
    funktion_id: int
    funktion_name: str
    summe_stunden: float
    schwellenwert_stunden: float
    schwellenwert_ueberschritten: bool


class SchwellenwertEintragOut(BaseModel):
    person_id: int
    person_name: str
    funktion_id: int
    funktion_name: str
    summe_stunden: float
    schwellenwert_stunden: float
    uebernommen_stunden: float
    ueberschuss_stunden: float


class UebernahmeAnlegen(BaseModel):
    person_id: int
    funktion_id: int
    stunden: float = Field(gt=0)

File: frontend/src/api/dienststunden.ts
import { apiGet, apiPost } from "./client";
import type { DienststundenSummeOut } from "./types";

export interface DienststundenEintragOut {
  id: number;
  person_id: number;
  person_name: string;
  funktion_id: number;
  funktion_name: string;
  stunden: number;
  datum: string;
  ohne_pin: boolean;
}

export const stundenErfassen = (
  funktionId: number,
  stunden: number,
  datum: string,
  ohnePin = false
) =>
  apiPost<DienststundenEintragOut>("/dienststunden", {
    funktion_id: funktionId,
    stunden,
    datum,
    ohne_pin: ohnePin,
  });

export const holeMeineSummen = () =>
  apiGet<DienststundenSummeOut[]>("/dienststunden/meine");

export const dienststundenReservierungAnlegen = () =>
  apiPost<{ token: string; ablauf_am: string }>("/dienststunden/reservierung");
